### DISCUSSION TITLE
how to dynamically add marks to a plot?

### DISCUSSION THREAD

**Discussion by zillionare** [h2oai/wave#1483] [Q&A]
I'm drawing a backtest plot with live update. The basic marks are lines. In some special time, in case there's trade, I need mark a point on the plot, to indicate a `buy` or `sell`:
<img width="1120" alt="image" src="https://user-images.githubusercontent.com/7206696/172043882-b6e7872e-8ac7-4c35-82d1-20e7e8858c98.png">

I want add text annotations on every turning point in the above plot.

In other word, if your plot use circular data buffer, that means the data is changing as time passed, so how to add other marks so they can by synced/aligned with main marks?

**Comment by lo5** [comment]
The x-axis isn't visible in the screenshot, so i can't comment on whether circular buffers are the right choice here.

Circular buffers are mainly useful for temporal axes, where one wants to append a tick, and automatically drop the oldest tick. 

Ticks are appended with `[-1]`, like this:
```
my_buffer[-1] = [a, b, c, d]
```

Ref: https://wave.h2o.ai/docs/buffers#buffers-in-practice
